src/host/outputStream.cpp

```diff
@@ -237,43 +238,28 @@ bool ConhostInternalGetSet::SetConsoleWindowInfo(const bool absolute, const SMAL
 }
 
 // Routine Description:
-// - Connects the PrivateSetCursorKeysMode call directly into our Driver Message servicing call inside Conhost.exe
-//   PrivateSetCursorKeysMode is an internal-only "API" call that the vt commands can execute,
+// - Sets the various terminal input modes.
+//   SetInputMode is an internal-only "API" call that the vt commands can execute,
 //     but it is not represented as a function call on out public API surface.
 // Arguments:
-// - fApplicationMode - set to true to enable Application Mode Input, false for Normal Mode.
+// - mode - the input mode to change.
+// - enabled - set to true to enable the mode, false to disable it.
 // Return Value:
-// - true if successful (see DoSrvPrivateSetCursorKeysMode). false otherwise.
-bool ConhostInternalGetSet::PrivateSetCursorKeysMode(const bool fApplicationMode)
-{
-    return NT_SUCCESS(DoSrvPrivateSetCursorKeysMode(fApplicationMode));
-}
-
-// Routine Description:
-// - Connects the PrivateSetKeypadMode call directly into our Driver Message servicing call inside Conhost.exe
-//   PrivateSetKeypadMode is an internal-only "API" call that the vt commands can execute,
-//     but it is not represented as a function call on out public API surface.
-// Arguments:
-// - fApplicationMode - set to true to enable Application Mode Input, false for Numeric Mode.
-// Return Value:
-// - true if successful (see DoSrvPrivateSetKeypadMode). false otherwise.
-bool ConhostInternalGetSet::PrivateSetKeypadMode(const bool fApplicationMode)
+// - true if successful. false otherwise.
+bool ConhostInternalGetSet::SetInputMode(const TerminalInput::Mode mode, const bool enabled)
 {
-    return NT_SUCCESS(DoSrvPrivateSetKeypadMode(fApplicationMode));
-}
+    auto& terminalInput = _io.GetActiveInputBuffer()->GetTerminalInput();
+    terminalInput.SetInputMode(mode, enabled);

```

Comment:
Why did the old code use `ServiceLocator::LocateGlobals()` for this?
I'm assuming that was just some code cruft / accidental roundaboutness?

Reply:
Well the original code was calling through to static functions in getset.cpp which aren't part of the `ConhostInternalGetSet` class, and so the only way they can access that state was through `ServiceLocator::LocateGlobals`. I think this was originally because the code in getset.cpp was mostly implementations of the public console APIs, and the same pattern was followed when private functions were added. This has resulted in a fair amount of unnecessary boilerplate code, which I'm hoping we can cut down on over time.